src/iface-modbus.c

```diff
@@ -194,7 +195,58 @@ static void attempt_connect(struct l_timeout *to, void *user_data)
 	l_timeout_modify(to, RECONNECT_TIMEOUT);
 }
 
-int iface_modbus_read_data(int reg_addr, int bit_offset, knot_value_type *out)
+static void iface_modbus_config_endiannes_type_recv_32_bits(uint32_t *src,
+						int endianness_type)
+{
+	switch (endianness_type) {
+	case MODBUS_ENDIANNESS_TYPE_BIG_ENDIAN:
+		*src = ((*src & 0xffff0000u) >> 16)|
+			((*src & 0x0000ffffu) << 16);
+		break;
+	case MODBUS_ENDIANNESS_TYPE_MID_BIG_ENDIAN:
+		*src = ((((*src) & 0xff000000u) >> 24)|
+			(((*src) & 0x00ff0000u) >> 8)|
+			(((*src) & 0x0000ff00u) << 8)|
+			(((*src) & 0x000000ffu) << 24));
+		break;
+	case MODBUS_ENDIANNESS_TYPE_LITTLE_ENDIAN:
+		*src = ((((*src) & 0xff000000u) >> 8)|
+			(((*src) & 0x00ff0000u) << 8)|
+			(((*src) & 0x0000ff00u) >> 8)|
+			(((*src) & 0x000000ffu) << 8));
+		break;
+	case MODBUS_ENDIANNESS_TYPE_MID_LITTLE_ENDIAN:
+		break;
+	default:
+		break;
+	}
+}
+
+static void iface_modbus_config_endiannes_type_recv_64_bits(uint64_t *src,
+						int endianness_type)
+{
+	if (endianness_type == MODBUS_ENDIANNESS_TYPE_MID_BIG_ENDIAN ||
```

Comment:
Yes, for mid-big endian it's necessary to go through the two ifs

Reply:
I do not think it is necessary for both conditions since the second one will override the first if the "endianness_type" is "MODBUS_ENDIANNESS_TYPE_LITTLE_ENDIAN".